packages/headless/src/features/recent-queries/recent-queries-slice.test.ts

```diff
@@ -134,4 +134,22 @@ describe('recent-queries slice', () => {
       testQueries
     );
   });
+
+  it('should not add new recent query on search fulfilled if queue already contains the query with extra whitespace', () => {
```

Comment:
I think you could have piggy backed on the previous test with multiple input case.

```
it('should not add new recent query on search fulfilled if queue already contains the query', () => {
    const duplicates = ['what is a query', '   what is a query', '    what is a query '];
    for(const otherTestQuery in duplicates) {
      [...]
   }
});
````